.github/workflows/auto-upload-image.yml

```diff
@@ -0,0 +1,32 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Java-CI-for-auto-upload-docker-image
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  build:
+    env:
+      USERNAME: smaks
+      IMAGE_NAME: gateway
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 15
+        uses: actions/setup-java@v2
+        with:
+          java-version: '15'
+          distribution: 'adopt'
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Build with Gradle
+        run: ./gradlew build
+      - name: Build image
+        run: docker build -t $USERNAME/$IMAGE_NAME:$(./gradlew version -q) .
+      - name: Authorization Docker Hub
+        run: docker login -u $USERNAME -p '${{secrets.DOCKERHUB_PASSWORD}}'
```

Comment:
Look at this action https://github.com/marketplace/actions/docker-login

Reply:
Or even this one https://github.com/marketplace/actions/build-and-push-docker-images